### ISSUE TITLE
Shift + Space extrude broken in 4.0

### ISSUE THREAD

**Issue by alziepen** [HEAVYPOLY/HEAVYPOLY_Blender#23]
[HP shiftspace broken.txt](https://github.com/HEAVYPOLY/HEAVYPOLY_Blender/files/12373069/HP.shiftspace.broken.txt)
Shift + Space extrude hotkey broken

**Comment by moooncube** [comment]
I had the same problem, can confirm.
The issue seems to be that face maps have been removed in 4.0, as listed here in the "Mesh" section. That breaks the `bpy.ops.object.face_map_add` operator and causes the error you shared

https://wiki.blender.org/wiki/Reference/Release_Notes/4.0/Python_API

My workaround that solved the problem is to go into Edit -> Preferences -> Keymap, search for "Context Sensitive Extrude" (which starts the HP script) and disable both entries using the checkboxes.
Then search for "Extrude and Move on Normals" and set the keyboard shortcut to what you're used to, probably Shift + Spacebar.

**Comment by doommchips** [comment]
@moooncube would you be able to push a fix to the error?

**Comment by moooncube** [comment]
@doommchips No unfortunately not, since my “fix“ is just reassigning some shortcuts to not trigger the problem, but it‘s still there regardless. I don‘t even know if Vaughan is still around on here to update things, I think he‘s mostly focused on Heavypaint these days.
Plus: my solution I posted above seems to not work with 4.1 anymore.

BUT! Somebody on the Heavypoly Discord server figured out a more permanent fix that also works with 4.1 for me:

Go into the scripts folder and open up the file called HEAVYPOLY_OPERATORS.py
Replace everything in it with the contents of this attached file and it should work again as expected.

Note: Don‘t just drop the .txt file in there because that won‘t do anything, it needs to be in the .py file. I just had to upload as .txt because it won‘t allow .py files.

[HEAVYPOLY_OPERATORS.txt](https://github.com/HEAVYPOLY/HEAVYPOLY_Blender/files/13194385/HEAVYPOLY_OPERATORS.txt)
